packages/driver/src/conUtils.ts

```diff
@@ -694,13 +694,26 @@ async function parseConnectDsnAndArgs(
           .catch(() => undefined),
       ]);
 
+      let database = _database;
+
+      if (database !== undefined && branch !== undefined) {
+        if (database !== branch) {
+          throw new InterfaceError(
+            "Both database and branch exist in the config dir and don't match.",
```

Comment:
Do we have a shared client test case for this?

Reply:
I don't think that we have for this specific case, but possibillity for this is very rare.

Reply:
yeah, the test cases cover (currently) rare or impossible cases just to make sure all clients treat different edge cases the same way. probably worth adding a case!